Update README.md with release installation guide (#202)
kyma-project/istio: README.md

The Istio module allows you to add Istio Operator to the Kyma runtime. Within Istio Operator, Istio Controller is responsible for installing, uninstalling, and managing Istio. For more information, read the [Istio Controller documentation](./docs/user/00-10-overview-istio-controller.md).

## Install Istio Manager and Istio from the latest release

### Prerequisites

- Access to a Kubernetes (v1.24 or higher) cluster
- [kubectl](https://kubernetes.io/docs/tasks/tools/)

### Procedure

> 1. Clone the project and open the `istio` folder.

Reviewer: I don't know why we should clone the project? It is not used or am I missing something?

Author: We do not need since we will not be deploying the sample CR but rather apply the release CR from assets.

   ```bash
   git clone https://github.com/kyma-project/istio.git && cd istio/
   ```

2. To install Istio, you must install latest Istio Manager first. Run the following:

   ```bash

>    kubectl create ns kyma-system

Reviewer: As far as I know the namespace is included in the `istio-manager.yaml` asset.

Author: Yes, you are correct: https://github.com/kyma-project/istio/blob/18ae0c476fd96310cefc542cb6b4b043a459d50e/config/default/kustomization.yaml#L1-L2

   kubectl apply -f https://github.com/kyma-project/istio/releases/latest/download/istio-manager.yaml
   ```

3. To get Istio installed, apply the latest Istio CR:

   ```bash

>    kubectl apply -f https://github.com/kyma-project/istio/releases/latest/download/istio-default-cr.yaml

Reviewer: The name of the asset in the release job is different. I don't know what name we should use:
https://github.com/kyma-project/istio/blob/eba042a3dd45dcd8bcf6bf5c0614f88a475e6e84/scripts/publish_assets.sh#L35

Author: Will rename the file to `istio-default-cr.yaml` it is more descriptive of what the file is.

The Istio module allows you to add Istio Operator to the Kyma runtime. Within Istio Operator, Istio Controller is responsible for installing, uninstalling, and managing Istio. For more information, read the [Istio Controller documentation](./docs/user/00-10-overview-istio-controller.md).

## Install Istio Manager and Istio from the latest release

### Prerequisites

- Access to a Kubernetes (v1.24 or higher) cluster
- [kubectl](https://kubernetes.io/docs/tasks/tools/)

### Procedure

> 1. To install Istio, you must install latest Istio Manager first. Run the following:

Reviewer 2: The file contains both Deployment, CRD and RBAC for the controller. Maybe the name of the file could also be adjusted?
```suggestion
1. To install Istio, you must install latest Istio Manager and Istio CRD first. Run the following:
```

Author: hmm, they are all needed for the `istio-manager` as whole component entity. do you have better name for the file?

Reviewer 2: If you are fine with the name it's OK, I would only update the documentation then that it contains all the needed stuff

Author: Personally I prefer `istio-operator.yaml`, but to align to KEDA and other we should stick to manager I guess.

   ```bash
   kubectl apply -f https://github.com/kyma-project/istio/releases/latest/download/istio-manager.yaml
   ```

2. To get Istio installed, apply the latest default Istio CR:

   ```bash
   kubectl apply -f https://github.com/kyma-project/istio/releases/latest/download/istio-default-cr.yaml
   ```

   You should get a result similar to this example:

   ```bash
   istio.operator.kyma-project.io/default created
   ```

3. Check Istio CR state to verify a successful installation with:

   ```bash
   kubectl get -n kyma-system istios/default
   ```

>    After successful installation you should get `Ready` state:

Reviewer 3: ```suggestion
   After successful installation, you get the following output:
```

   ```bash
   NAME      STATE
   default   Ready
   ```

> For more installation options, visit [Install Istio Manager](./docs/contributor/01-00-installation.md)

Reviewer 3: ```suggestion
For more installation options, visit [Install the Istio module](./docs/contributor/01-00-installation.md).
```
Our installation doc is named 'Install the Istio module', but maybe the name 'Install Istio Operator' would be better. What do you think?
